packages/server-wallet/src/chain-service/chain-service.ts

```diff
@@ -347,7 +359,7 @@ export class ChainService implements ChainServiceInterface {
         this.addFinalizingChannel(channelId, finalizesAt);
```

Comment:
Is this block of code for the case [when a new challenge overwrites an existing one](https://docs.statechannels.org/docs/protocol-tutorial/clear-a-challenge#call-challenge-again)? 

 Does it assume that this new `finalizesAt` is later than the one we had before? I think that is not a completely safe assumption. It is a bit of an edge case, but it is possible for the `challengeDuration` of a subsequent challenge to be shorter than for the original challenge, and therefore for the `finalizesAt` to decrease.

In the case where the new `finalizesAt` is in fact earlier, we will end up with the new entry in slot 0, after we do `addFinalizingChannel`, and then the new entry will be sliced off (instead of the current one) and we are in an infinite loop. 

It does feel a bit like we are possibly fighting against the wrong data structure. See my other comment.

Reply:
I think you are right, there is an issue here with a potential infinite loop. Great catch!

Reply:
I _think_ might be resolved by 49513416a.